java/ql/src/Telemetry/AutomodelApplicationModeExtractCandidates.ql

```diff
@@ -55,7 +55,7 @@ private Endpoint getSampleForSignature(
 from
   Endpoint endpoint, string message, ApplicationModeMetadataExtractor meta, DollarAtString package,
   DollarAtString type, DollarAtString subtypes, DollarAtString name, DollarAtString signature,
-  DollarAtString input, DollarAtString isVarargsArray
+  DollarAtString input, DollarAtString isVarargsArray, DollarAtString alreadyAiModeled
```

Comment:
I wonder if it would be easier to just select the provenance (or an empty string if no provenance exists), and then handle the filtering on the backend. This would be slightly more flexible in case we want compare with non-`ai-` provenances.

Reply:
I'm adding the provenance, but only for provenances matching `ai-%`. The other provenances are likely going to remain uninteresting for us (and if not, easily changeable).

Reply:
[808dc3e](https://github.com/github/codeql/pull/13954/commits/808dc3e8d306dcb6441f27bb3c322917d978c8d6)